dlt/destinations/impl/lancedb/exceptions.py

```diff
@@ -0,0 +1,62 @@
+from functools import wraps
+from typing import (
+    Any,
+)
+
+from lancedb.exceptions import MissingValueError, MissingColumnError  # type: ignore[import-untyped]
+from pyarrow import ArrowInvalid
+
+from dlt.common.destination.exceptions import (
+    DestinationUndefinedEntity,
+    DestinationTerminalException,
+    DestinationException,
+    DestinationTransientException,
+)
+from dlt.common.destination.reference import JobClientBase
+from dlt.common.typing import TFun
+
+
+class LanceDBBatchError(DestinationException):
+    pass
+
+
+def lancedb_error(f: TFun) -> TFun:
```

Comment:
I thought I left a comment about this somewhere: Why can we not use the same error handling mechanism we use on all the other destinations: @raise_database_error?

Reply:
@sh-rp I don't think it will work because lancedb doesn't have a DBAPI driver we can use, i.e. doesn't inheret from SqlClientBase

Reply:
Let me know if I'm missing something?

Reply:
Ah yes, you are right, we should make this decorator more universal, forget about it for now ;)